tools/esp-api/README.md

```diff
@@ -0,0 +1,39 @@
+# Developer README for ESP API Command Line Tool
+
+## Overview
+
+This tool is designed to interact with HPCC ESP services, providing four commands: `describe`, `test`, `list-services`, and `list-methods`.
+
+- **`describe`:**
+  Allows users to explore available services, methods, and their request-response structures.
+
+- **`test`:**
+  Enables sending requests in various formats (XML, JSON, or form query strings) to the ESP services.
+
+- **`list-services`:**
+  Invoked by `esp-api list-services`, this command provides a list of names of all ESP services. This output is used by the auto-complete script.
```

Comment:
Choose between:
1. Each command is invoked by `esp-api <command>`.
2. Invoked by the auto-complete script. The last sentence would then be redundant.

I would pick #2.